Base/Python/slicer/__init__.py

```diff
@@ -1,4 +1,108 @@
-""" This module sets up root logging and loads the Slicer library modules into its namespace."""
+""" This module sets up root logging and loads the Slicer library modules into its namespace.
+
+
+.. py:attribute:: app
+
+  This is set to the singleton instance of ``qSlicerApplication``::
+
+    >>> slicer.app
+    qSlicerApplication (qSlicerApplication at: 0x7ffd066a07a0)
+
+  .. warning::
+
+    This attribute is set only in the context of a running Slicer application.
+
+
+.. py:attribute:: mrmlScene
+
+  This is set to the instance of the MRML Scene::
+
+      >>> slicer.mrmlScene
+      <MRMLCore.vtkMRMLScene(0x2797cb0) at 0x7fbfedb82520>
+
+      >>> slicer.mrmlScene == slicer.app.mrmlScene()
+      True
+
+  .. warning::
+
+    This attribute is set only in the context of a running Slicer application.
+
+
+.. py:attribute:: modules
+
+  This object provides access to all instantiated Slicer modules.
+
+  For each instantiated Slicer module, an attribute named after the lower-cased
+  Slicer module name (``slicer.module.<modulename>``) is associated with the
+  corresponding instance of ``qSlicerAbstractCoreModule``.
+
+  For example::
+
+    >>> slicer.modules.volumes
+    qSlicerVolumesModule (qSlicerVolumesModule at: 0x4869000)
+
+    >>> slicer.modules.volumes.inherits("qSlicerAbstractModule")
+    True
+
+    >>> slicer.modules.volumes.inherits("qSlicerLoadableModule")
+    True
+
+  .. warning::
+
+    These attributes are set only in the context of a running Slicer application.
+
+  Additionally for all scripted modules (`qSlicerScriptedLoadableModule`), these
+  additional attributes are also set:
+
+  * the attribute ``<moduleName>Instance`` is set to the corresponding instance
+    of :class:`~slicer.ScriptedLoadableModule.ScriptedLoadableModule`.
+
+    For example::
+
+      >>> slicer.modules.VectorToScalarVolumeInstance
+      <VectorToScalarVolume.VectorToScalarVolume object at 0x7fbfc1d1aa60>
+
+      >>> isinstance(slicer.modules.VectorToScalarVolumeInstance, slicer.ScriptedLoadableModule.ScriptedLoadableModule)
+      True
+
+  * the attribute ``<moduleName>Widget`` is set to the corresponding instance
```

Comment:
What if the Python module widget is reloaded? Does it point to the current widget instance or the first? The first module widget may become obsolete if the module widget is reloaded.